### PR TITLE
Case start_date/end_date are not datetime

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
`start_date` and `end_date` are not datetime fields but the schema specifies they are datetime. This means that formbuilder incorrectly displays them as datetime.

Before
----------------------------------------
Incorrectly datetime.

After
----------------------------------------
Correctly date.

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Case/DAO/Case.php [hunk 1]
```diff
@@ -260,7 +260,7 @@ public static function &fields() {
           'localizable' => 0,
           'html' => [
             'type' => 'Select Date',
-            'formatType' => 'activityDateTime',
+            'formatType' => '`activityDate',
```
Comment (yashodha) [inline CRM/Case/DAO/Case.php:263]: @mattwire looks like a stray ` in` '`activityDate',`

File: xml/schema/Case/Case.xml [hunk 2]
```diff
@@ -79,7 +79,7 @@
     <comment>Date on which given case starts.</comment>
     <html>
       <type>Select Date</type>
-      <formatType>activityDateTime</formatType>
+      `<formatType>`activityDate</formatType>
```
Comment (yashodha) [inline xml/schema/Case/Case.xml:82]: looks like a stray `s again

File: CRM/Case/DAO/Case.php (hunk 1 above)
Comment (mattwire) [reply CRM/Case/DAO/Case.php]: Fixed thanks

File: xml/schema/Case/Case.xml (hunk 2 above)
Comment (mattwire) [reply xml/schema/Case/Case.xml]: Fixed thanks

File: xml/schema/Case/Case.xml [hunk 3]
```diff
@@ -79,7 +79,7 @@
     <comment>Date on which given case starts.</comment>
     <html>
       <type>Select Date</type>
-      <formatType>activityDateTime</formatType>
+      `<formatType>activityDate</formatType>
```
Comment (seamuslee001) [inline xml/schema/Case/Case.xml:82]: stray ` at the start of this line @mattwire

Comment (mattwire) [reply xml/schema/Case/Case.xml]: Oh for goodness sake...

**Comment by demeritcowboy** [comment]
I'm not sure I understand the PR? On open case there is a time component. And start date is conceptually identical to the activity date(time) on open case. It's just for some reason exists as its own field in the database. (In fact all "case level" fields are theoretically computable from the activity sequence - but I'm getting off topic).

Also there are some missing backticks ... kidding!

**Comment by mattwire** [comment]
@demeritcowboy This just fixes the xml schema to match the actual format in the database. The case table has 4 dates: created_date, modified_date which are timestamp and start_date,end_date which are date only.
I agree that there is probably some duplication between open case activity and case table itself.

**Comment by demeritcowboy** [comment]
Ok I see. And I see formbuilder even before the PR ignores whatever time you put when creating the open case activity, which is different from the regular new case form. Hmm, awkward.

**Comment by demeritcowboy** [comment]
I think I regret merging this. Getting rid of these fields is too much of an uphill battle, but these should actually be datetimes (with timezone) or timestamps because date-only fields assume all cases are in the same timezone, and also this is now pushes the mismatch with activity datetime further.

I don't have the bandwidth for it but my vote would be that the fields in the db should be converted and form builder should be updated to not ignore the time component.

**Pull request merged**